independent-projects/tools/registry-descriptor/src/main/java/io/quarkus/registry/model/Release.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.registry.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonDeserialize(as = ImmutableRelease.class)
+public interface Release extends Comparable<Release> {
+
+    String getVersion();
+
+    @Nullable
+    @JsonProperty("quarkus-core")
+    String getQuarkusCore();
+
+    @JsonProperty("repository-url")
+    @Value.Auxiliary
+    @Nullable
+    String getRepositoryURL();
+
+    static ImmutableRelease.Builder builder() {
+        return ImmutableRelease.builder();
+    }
+
+    @Override
+    default int compareTo(Release o) {
+        //TODO: Compare using SemVer rules
```

Comment:
Hint: org.apache.maven.artifact.versioning.DefaultArtifactVersion

Reply:
Thanks! That class uses `org.apache.maven.artifact.versioning.ComparableVersion`, which I think is more appropriate. I'll use that :wink: